homeassistant/components/asuswrt/router.py

```diff
@@ -61,6 +59,20 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def _get_dict(keys: list, values: list) -> dict[str, Any]:
+    """Create a dict from a list of keys and values."""
+    ret_dict: dict[str, Any] = dict.fromkeys(keys)
+    if not values:
+        return ret_dict
+    if len(values) < len(keys):
+        return ret_dict
```

Comment:
When do these cases happen?

Reply:
Potentially never, it is just to manage a worst cases

Reply:
The only reason to have a case like this is that library return results different than expected. So you mean that in this cases is better to have an exception?

Reply:
Yes, if something unexpected happens, we want to know about it immediately so we can adjust our expectations and then adjust the code.